src/parser/node.js

```diff
@@ -14,8 +14,13 @@ class Node implements NodeBase {
     this.type = "";
     this.start = pos;
     this.end = 0;
-    this.loc = new SourceLocation(loc);
```

Comment:
^ in loose mode it's going to compile the class into a function anyway. @jridgewell said it's not worth it and it seems to not big that big a when anyway and to focus on the object shape instead